### PR TITLE
fix return staking total

### PR DESCRIPTION
**Pull request by dustinxie**

### CODE REVIEW

File: action/protocol/staking/bucket_pool.go [hunk 1]
```diff
@@ -128,13 +130,27 @@ func NewBucketPool(sr protocol.StateReader) (*BucketPool, error) {
...
-func (bp *BucketPool) Total() *big.Int {
-	return new(big.Int).Set(bp.total.amount)
+func (bp *BucketPool) Total(sr protocol.StateReader) (*big.Int, uint64, error) {
+	bp.lock.RLock()
+	defer bp.lock.RUnlock()
+
+	height, err := sr.Height()
+	if err != nil {
+		return nil, 0, err
+	}
```
Comment (CoderZhi) [inline action/protocol/staking/bucket_pool.go:140]: what's the value of reading height from sr here? unify api?

File: action/protocol/staking/bucket_pool_test.go [hunk 2]
```diff
@@ -66,8 +66,12 @@ func TestBucketPool(t *testing.T) {
 
 	pool, err := NewBucketPool(sm)
 	r.NoError(err)
-	r.Equal(big.NewInt(0), pool.Total())
-	r.EqualValues(0, pool.Count())
+	total, _, err := pool.Total(sm)
```
Comment (CoderZhi) [inline action/protocol/staking/bucket_pool_test.go:69]: how could we passing sm to both NewBucketPool and Total

Comment (dustinxie) [reply action/protocol/staking/bucket_pool_test.go]: fixed

File: action/protocol/staking/bucket_pool.go (hunk 1 above)
Comment (dustinxie) [reply action/protocol/staking/bucket_pool.go]: pls check latest commit

File: action/protocol/staking/candidate_statemanager.go [hunk 3]
```diff
@@ -14,20 +14,14 @@ import (
...
-		// TODO: remove CandidateCenter interface, return *candCenter
-		CandCenter() CandidateCenter
-		BucketPool() *BucketPool
-	}
-
 	// CandidateStateManager is candidate state manager on top of StateManager
 	CandidateStateManager interface {
 		protocol.StateManager
+		CandCenter() CandidateCenter
+		BucketPool() *BucketPool
```
Comment (CoderZhi) [inline action/protocol/staking/candidate_statemanager.go:24]: If we do have these two APIs, we may delete all the following APIs, except "Commit"

File: action/protocol/staking/candidate_statereader.go [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+	"github.com/pkg/errors"
+)
+
+type (
+	// CandidateStateReader contains candidate center and bucket pool
+	CandidateStateReader interface {
+		// TODO: remove CandidateCenter interface, return *candCenter
+		Height() uint64
+		CandCenter() CandidateCenter
+		BucketPool() *BucketPool
```
Comment (CoderZhi) [inline action/protocol/staking/candidate_statereader.go:21]: If we do return CandidateCenter and BucketPool, CandidateStateReader is not a real reader.

File: action/protocol/staking/candidate_statemanager.go (hunk 3 above)
Comment (dustinxie) [reply action/protocol/staking/candidate_statemanager.go:24]: yes, or we can delete these 2 API

**Pull request merged**